codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -396,18 +420,30 @@ public ChannelFuture writeRstStream(ChannelHandlerContext ctx, int streamId, lon
      * Sends a {@code GO_AWAY} frame to the remote endpoint and updates the connection state appropriately.
      */
     @Override
-    public ChannelFuture writeGoAway(ChannelHandlerContext ctx, int lastStreamId, long errorCode, ByteBuf debugData,
-            ChannelPromise promise) {
-        Http2Connection connection = connection();
+    public ChannelFuture writeGoAway(ChannelHandlerContext ctx, final int lastStreamId, final long errorCode,
+            ByteBuf debugData, ChannelPromise promise) {
+        final Http2Connection connection = connection();
         if (connection.isGoAway()) {
             debugData.release();
-            return ctx.newSucceededFuture();
+            return promise.setFailure(connectionError(INTERNAL_ERROR,
```

Comment:
Given the recent revelations of the spec clarifications I think this is just wrong in general :)  We should allow sending multiple go aways, as long as the `last stream id` is decreasing.